app/store/middleware.js

```diff
@@ -357,15 +357,19 @@ function cleanupState(action, keepCurrent = false) {
 export function shareExtensionData() {
     return (next) => (action) => {
         // allow other middleware to do their things
-        const nextAction = next(action);
-
-        switch (action.type) {
-        case UserTypes.LOGOUT_SUCCESS:
-            mattermostBucket.removePreference('emm', Config.AppGroupId);
-            mattermostBucket.removeFile('entities', Config.AppGroupId);
-            break;
+        if (typeof action === 'object' && action.type) {
+            const nextAction = next(action);
+
+            switch (action.type) {
+            case UserTypes.LOGOUT_SUCCESS:
+                mattermostBucket.removePreference('emm', Config.AppGroupId);
+                mattermostBucket.removeFile('entities', Config.AppGroupId);
+                break;
+            }
+            return nextAction;
         }
-        return nextAction;
+
+        return next();
```

Comment:
I think we should still pass `action` into `next` here

Reply:
if we do then the if above does not make any sense

Reply:
I don't think it makes sense to just call `next` without an argument. Then the next reducer will just receive `undefined` and possibly error because of that. If the other reducers weren't crashing before, I think it's better to just keep passing along whatever we've received